public/app/plugins/panel/graph/graph.ts

```diff
@@ -428,7 +452,9 @@ coreModule.directive('grafanaGraph', function($rootScope, timeSrv, popoverSrv) {
         let defaultTicks = panelWidth / 50;
 
         if (data.length && bucketSize) {
-          ticks = _.map(data[0].data, point => point[0]);
+          ticks = _.chain(data).map(d => {
+            return _.map(d.data, point => point[0]);
```

Comment:
why are you doing this? Also this looks pretty terrile from a performance standpoint. Never use map/each function callback looks when dealing with large value arrays, always prefer for loops (or for of loops)